### DISCUSSION TITLE
[Script request] Dawarich

### DISCUSSION THREAD

**Discussion by Aloe-recite** [community-scripts/ProxmoxVE#184] [Request script]
### Application Name

Dawarich

### Website

https://dawarich.app

### Description

Dawarich is a self-hosted web app designed to replace Google Timeline (aka Google Location History). It enables you to:
1. Visualize your data on an interactive map.
2. Import your location history from Google Maps Timeline and Owntracks.
3. Explore statistics like the number of countries and cities visited, total distance traveled, and more!

It's not something "necessary" for most of us, but indeed funny and useful.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by ptrsmk** [comment]
I'm curious how this compares to [OwnTracks](https://github.com/owntracks) and [Traccar](https://github.com/traccar).

**Reply by Aloe-recite** [reply]
I don't know since I didn't try them out. Traccar seems more oriented towards GPS devices logging or such (they say the sypoport more tna 200 protocols so mayber it's more oriented towards this), OwnTracks not so sure. However, Dawarich is frequently mentioned on Reddit, YT and the like, whereas OwnTracks I never heard of it before, but looks interesting too of course - I'm not making a comparison, just personal statistics

**Reply by Freika** [reply]
@ptrsmk Dawarich was created out of my own need to have a more flexible solution to visualize my location history. I previously was using Owntracks, and it felt like it doesn't have features I wanted and felt necessary. They both solves the same problem, but Dawarich provides more features.

**Comment by Oberlandix** [comment]
Dear all! Is there any hope for a "ready" installation-script? 
"dawarich" is more and more known and I did hear a lot´s of podcasts related to this app in the past.
The App "Immich" would be a very good companion here, too.

Looking forward to hear any news about this topic (I am not a professionell in creating frist a docker-container on proxmox... sorry!).

**Comment by heidle78** [comment]
I would use it directly, waiting also for a proxmox solution. 
In a german IT magazine it got just positive feedback - can´t await to use it via proxmox - hopefully we see a script very soon.

link would be: https://www.heise.de/ratgeber/Raspi-Privaten-Standortverlauf-mit-Dawarich-lokal-aufzeichnen-10235624.html

**Comment by neoscaler** [comment]
As dawarich is Docker Compose based, there is no need for a Proxmox helper script. 

Just spin up a new Docker LXC container (via https://community-scripts.github.io/ProxmoxVE/scripts?id=docker for example), download the compose file in the shell and start it via `docker compose up -d`.

**Comment by marc50ca** [comment]
I've used Dawarich of quite some time and it works well, however there are some updates where the developer has made changes that require steps to be taken before upgrading (moving from Sqlite to Postgres, away from Redis the back to Redis)

And yes I've found out the hard way by not reading release notes etc first.

So while it could go in a community script loading new version shouldn't be handle by the update utility.

**Comment by MickLesk** [comment]
I've already struggled with the script twice, and somehow never got it to work, sometimes I can't figure out the Dockerfile either. I think that I need support from @Freika to build a Debian installer. (Community scripts and maybe for his project itself)

**Reply by Freika** [reply]
How can I help?

**Reply by MickLesk** [reply]
I can share you next week my First Prototype, I think somewhere in the backend (ruby) it fails, maybe I have overlooked something several times

**Reply by bachjessen** [reply]
@MickLesk sorry to tag you! did you end up having it run with help from Freika?

**Comment by RiX012** [comment]
Hi, any word on if this is going to work? I would love to run Dawarich directly in an LXC!

**Comment by MickLesk** [comment]
https://github.com/community-scripts/ProxmoxVED/issues/1307

Ready for Testing

**Comment by hashblen** [comment]
With the current community script,  I had an issue with the nginx config.

The Nginx config serves `/assets` with `try_files $uri =404`. This prevents Rails from handling some asset module requests. For my setup, the resulting 404 is returned as HTML, causing browser MIME errors (`text/html` instead of JavaScript).

Removing the `/assets` location block fixed the issue, making it closer to the doc's example https://dawarich.app/docs/self-hosting/configuration/reverse-proxy/#nginx without brotli.

(my `/etc/nginx.sites-available/dawarich.conf` is then just the following:)
```nginxconf
upstream dawarich {
    server 127.0.0.1:3000;
}

server {
    listen 80;
    server_name _;

    client_max_body_size 100M;

    location / {
        proxy_pass http://dawarich;
        proxy_http_version 1.1;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_redirect off;
        proxy_buffering off;
    }
}
```
